plugins/extractors/maxcompute/maxcompute.go

```diff
@@ -0,0 +1,251 @@
+
+	var marker string
+	var counter int
+	for {
+		e.logger.Info("fetching tables", "marker", marker)
+		resp, err := e.fetchTables(apiClient, marker)
+		if err != nil {
+			return err
+		}
+
+		counter += len(resp.Body.Data.Tables)
+		e.logger.Info("fetched tables", "count", counter)
+		wg.Add(len(resp.Body.Data.Tables))
+
+		for _, table := range resp.Body.Data.Tables {
+			go e.processTable(apiClient, table, emit, &wg)
+		}
+		wg.Wait()
+		if len(resp.Body.Data.Tables) == 0 || len(resp.Body.Data.Tables) < int(e.config.MaxPageSize) || *resp.Body.Data.Marker == "" {
+			break
+		}
+		marker = *resp.Body.Data.Marker
+	}
+
+	return nil
+}
+
+func (e *Extractor) fetchTables(apiClient *maxcomputeclient.Client, marker string) (*maxcomputeclient.ListTablesResponse, error) {
+	return apiClient.ListTables(&e.config.ProjectName, &maxcomputeclient.ListTablesRequest{
+		MaxItem: &e.config.MaxPageSize,
+		Marker:  &marker,
+	})
+}
+
+func (e *Extractor) processTable(apiClient *maxcomputeclient.Client, table *maxcomputeclient.ListTablesResponseBodyDataTables,
+	emit plugins.Emit, wg *sync.WaitGroup,
+) {
+	defer wg.Done()
+
+	tableInfo, err := apiClient.GetTableInfo(&e.config.ProjectName, table.Name, &maxcomputeclient.GetTableInfoRequest{})
+	if err != nil {
+		e.logger.Error("failed to get table info", "table", *table.Name, "error", err)
+		return
+	}
+
+	asset, err := e.buildAsset(tableInfo)
+	if err != nil {
+		e.logger.Error("failed to build asset", "table", *table.Name, "error", err)
+		return
+	}
+
+	emit(models.NewRecord(asset))
+}
+
+func (e *Extractor) buildAsset(tableInfo *maxcomputeclient.GetTableInfoResponse) (*v1beta2.Asset, error) {
+	defaultSchema := "default"
+	if tableInfo.Body.Data.Schema == nil {
+		tableInfo.Body.Data.Schema = &defaultSchema
+	}
+
+	tableURN := plugins.MaxComputeURN(*tableInfo.Body.Data.ProjectName, *tableInfo.Body.Data.Schema, *tableInfo.Body.Data.Name)
+
+	asset := &v1beta2.Asset{
+		Urn:         tableURN,
+		Name:        *tableInfo.Body.Data.Name,
+		Type:        *tableInfo.Body.Data.Type,
+		Description: *tableInfo.Body.Data.Comment,
+		CreateTime:  convertInt64ToTimestamp(tableInfo.Body.Data.CreationTime),
+		UpdateTime:  convertInt64ToTimestamp(tableInfo.Body.Data.LastModifiedTime),
+		Service:     "maxcompute",
+	}
+
+	attributesData := buildAttributesData(tableInfo)
+
+	var columns []*v1beta2.Column
+	for _, col := range tableInfo.Body.Data.NativeColumns {
+		columnData := &v1beta2.Column{
+			Name: *col.Name,
+		}
+		columns = append(columns, columnData)
+	}
+
+	tableData := &v1beta2.Table{
+		Attributes: utils.TryParseMapToProto(attributesData),
+		Columns:    columns,
+		UpdateTime: convertInt64ToTimestamp(tableInfo.Body.Data.LastDDLTime),
+	}
+
+	table, err := anypb.New(tableData)
+	if err != nil {
+		e.logger.Warn("error creating Any struct", "error", err)
+	}
+	asset.Data = table
+
+	return asset, nil
+}
+
+func buildAttributesData(tableInfo *maxcomputeclient.GetTableInfoResponse) map[string]interface{} {
```

Comment:
please check if we can get the labels as well

Reply:
We can get it, one of the approach is from `table.Schema()` and get `TableLabel`
example code:
https://github.com/goto/meteor/pull/69/files#r1835980309

Reply:
referred the RFC for this, lets discuss

Reply:
addressed